Automattic/wp-calypso
# Purchases: Show new pre-cancellation form when cancelling non-plan products

Fixes https://github.com/Automattic/wp-calypso/issues/65186
Part of https://github.com/Automattic/wp-calypso/issues/56393


Please note that reviewing these changes is much easier by using this diff view:
<img width="150" alt="Screen Shot 2022-07-04 at 13 55 40" src="https://user-images.githubusercontent.com/1233880/177166392-f8f8875e-fa74-4818-b45b-7fd82142b3a5.png">

#### Proposed Changes

Displays the new pre-cancellation form introduced by https://github.com/Automattic/wp-calypso/pull/56206 when cancelling other products than plans, so we can show the Atomic revert warning before cancelling a marketplace plugin from an Atomic site with a Starter plan.

Before | After (step 1) | After (step 2)
--- | --- | ---
<img width="500" alt="Screen Shot 2022-07-04 at 10 05 38" src="https://user-images.githubusercontent.com/1233880/177111287-4607f8ac-cf6c-42e0-ac05-f5f189463494.png"> | <img width="500" alt="Screen Shot 2022-07-04 at 15 13 14" src="https://user-images.githubusercontent.com/1233880/177166208-52135b52-8c6b-4bfb-9001-4a524c84e593.png"> | <img width="500" alt="Screen Shot 2022-07-04 at 15 13 22" src="https://user-images.githubusercontent.com/1233880/177166234-8fe24d84-278f-4c90-9562-e67cd5696e32.png">


#### Testing Instructions

- Use the Calypso live link below
- Go to Plugins
- Purchase a paid plugin
- Go to Upgrades > Purchases
- Cancel the purchase
- Make sure you see the new cancellation form. In the first step you should be able to provide some feedback, and in the second step you should be warned about the Atomic revert.
- Make sure there are no regressions when cancelling a plan from a site (you should get the same cancellation form as in prod).
- Make sure you don't see the Atomic revert warning when cancelling a marketplace plugin from a site that has additional marketplace plugins
- Make sure you don't see the Atomic revert warning when cancelling other non-plan purchases that don't support the Atomic feature (e.g. the No Ads addon)
